flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorFailureTest.java

```diff
@@ -116,6 +122,8 @@ public void testFailingCompletedCheckpointStoreAdd() throws Exception {
 		verify(operatorSubtaskState.getRawOperatorState().iterator().next()).discardState();
 		verify(operatorSubtaskState.getManagedKeyedState().iterator().next()).discardState();
 		verify(operatorSubtaskState.getRawKeyedState().iterator().next()).discardState();
+		verify(operatorSubtaskState.getInputChannelState().iterator().next()).discardState();
+		verify(operatorSubtaskState.getResultSubpartitionState().iterator().next()).discardState();
```

Comment:
Agree, but this is an existing test which already uses mockito.
Having some fields mocked and some not would be confusing and changing the whole test is out of scope of this PR.

Reply:
I'm not asking to rewrite the whole test (although that would be nice ;) ). Having said that, I would prefer to have mixed test instead of increasing our dependency on the mockito, as this is adding more and more technological debt. 

Also we have lot's of existing tests that are mixing mockito mocked classes with proper classes, so I don't think this is an issue.

Reply:
Yes, I do have strong opposite feeling :) 
(I see only 2 new lines; and disadvantages of mixed tests).